backend/qemu.pm

```diff
@@ -1127,8 +1127,11 @@ sub handle_qmp_command ($self, $cmd, %optargs) {
         bmwqemu::fctinfo("Skipping the following qmp_command because QEMU_ONLY_EXEC is enabled:\n$line");
         return undef;
     }
-    my $wb = defined $optargs{send_fd} ? tinycv::send_with_fd($sk, $line, $optargs{send_fd}) : syswrite($sk, $line);
-    die "handle_qmp_command: syswrite failed $!" unless ($wb == length($line));
+    local $SIG{__DIE__} = undef;    # prevent custom die handler so that a broken pipe can be handled
+    my $wb = eval { defined $optargs{send_fd} ? tinycv::send_with_fd($sk, $line, $optargs{send_fd}) : syswrite($sk, $line) };
+    return {return => {status => 'shutdown'}} if $@ && $@ =~ qr/Broken pipe/;
+    die "handle_qmp_command: Unexpected error: $@" if $@;
+    die "handle_qmp_command: syswrite failed $!" unless $wb && $wb == length($line);
```

Comment:
> I would also save `$@` and `$!` in variables

This is how it could look like with Feature::Compat::Try:
```suggestion
    use Feature::Compat::Try;
    my $wb = do {
        try { defined $optargs{send_fd} ? tinycv::send_with_fd($sk, $line, $optargs{send_fd}) : syswrite($sk, $line) }
        catch ($e) { return {return => {status => 'shutdown'}} if $e =~ qr/Broken pipe/; die $e }
    };
    die "handle_qmp_command: syswrite failed $!" unless $wb && $wb == length($line);
```

Reply:
looks nice. I am preparing some changes to bring in Feature::Compat::Try and get rid of some older suboptimal code. I plan to continue here afterwards so that we have a good style for catching exception going forward before I introduce more as necessary for this PR.

Reply:
->
* #2650